include/nlohmann/detail/meta/cpp_future.hpp

```diff
@@ -147,8 +147,12 @@ struct static_const
     static constexpr T value{};
 };
 
-template<typename T>
-constexpr T static_const<T>::value; // NOLINT(readability-redundant-declaration)
+#ifndef JSON_HAS_CPP_17
```

Comment:
Why was this guard added?

Reply:
See d7954b29fcf1c0da74033435bdb82d24b6259cc1.

> Redelcaration of inline constexpr static data members in namespace scope
> was deprecated in C++17. Fixes -Werror=deprecated compilation failures.